app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -93,11 +93,12 @@ public void done(List<Page> objects, ParseException e) {
                     return;
                 }
                 for (Page page : objects) {
-                    if (page == null || page.getPageId() == null || page.getObjectId() == null) {
-                        Log.e("LoginActivity", "Page is null");
-                        return;
+                    if (page != null && page.getObjectId() == null && page.getPageId() != null) {
+                        existingPages.put(page.getPageId(), page.getObjectId());
+                    }
+                    else if (page == null){
```

Comment:
Lets move this up to do early return, and the next condition can be:
`if (page.getObjectId() != null && page.getPageId() != null)`